website/server/libs/auth/social.js

```diff
@@ -21,16 +24,49 @@ function _passportProfile (network, accessToken) {
   });
 }
 
+const applePrivateKey = nconf.get('APPLE_AUTH_PRIVATE_KEY');
+const applePublicKey = nconf.get('APPLE_AUTH_PUBLIC_KEY');
+
+const auth = new AppleAuth(JSON.stringify({
+  client_id: nconf.get('APPLE_AUTH_CLIENT_ID'), // eslint-disable-line camelcase
+  team_id: nconf.get('APPLE_TEAM_ID'), // eslint-disable-line camelcase
+  key_id: nconf.get('APPLE_AUTH_KEY_ID'), // eslint-disable-line camelcase
+  redirect_uri: 'https://habitica.com/api/v4/user/auth/social', // eslint-disable-line camelcase
+  scope: 'name email',
+}), applePrivateKey.toString(), 'text');
+
+async function _appleProfile (req) {
+  let idToken = {};
+  const code = req.body.code ? req.body.code : req.query.code;
+  const passedToken = req.body.id_token ? req.body.id_token : req.query.id_token;
+  if (code) {
+    const response = await auth.accessToken(code);
+    idToken = jwt.decode(response.id_token);
+  } else if (passedToken) {
+    idToken = await jwt.verify(passedToken, applePublicKey, { algorithms: ['RS256'] });
+  }
+  return {
```

Comment:
as discussed outside github there should be a check for when `idToken` doesn't have all the necessary data